config/initializers/controllers.rb

```diff
@@ -42,6 +42,7 @@ def registration
 
     return unless current_user.is_temp?
     store_url key: :registration_return_to
+    flash.keep
     redirect_to(current_user.is_email_pending? ? verification_sent_path : register_path)
```

Comment:
If the meaning of `is_email_pending?` is really `has_emails_and_all_unverified?`, this would mean that if a user has one verified email address and one unverified email address we won't let registration continue (but I think we should -- the minimum requirement is one email address verified).